migration-engine/core/src/commands/diagnose_migration_history.rs

```diff
@@ -132,22 +134,31 @@ impl<'a> MigrationCommand for DiagnoseMigrationHistoryCommand {
             .cloned()
             .collect();
 
-        let drift = match migration_inferrer.calculate_drift(&applied_migrations).await {
-            Ok(Some(rollback)) => Some(DriftDiagnostic::DriftDetected { rollback }),
-            Err(error) => Some(DriftDiagnostic::MigrationFailedToApply {
-                error: error.to_user_facing(),
-            }),
-            _ => None,
-        };
+        let (drift, error_in_unapplied_migration) = {
```

Comment:
How is this code behaving if we don't opt-in. Like, what is the difference between the options?

I don't know the feature, that's why asking.

Reply:
So if you don't opt-in, we won't try to detect drift (between expected state from your migraitons and actual state), and also won't apply unapplied migrations to a temporary database to validate them.

We made it opt-in so we don't create these temporary databases on presumably production databases.